AlphaGo/training/supervised_policy_trainer.py

```diff
@@ -745,77 +664,79 @@ def resume_training(args):
     else:
         raise ValueError("Metadata file not found!")
 
-    # determine what weight file to use
+    # Determine what weight file to use.
     if args.weights is None:
-        # newest epoch weight file from json
+        # Newest epoch weight file from json.
         weight_file = "weights.{epoch:05d}.hdf5".format(epoch=metadata["current_epoch"])
     else:
-        # user weight argument
+        # User weight argument.
         weight_file = args.weights
 
-    # check if training epochs has changed
+    # Check if training epochs has changed.
     if args.epochs is not None:
         metadata["epochs"] = args.epochs
 
     if args.verbose:
         print("trying to resume training from %s with weights %s" %
               (meta_file, os.path.join(args.out_directory, FOLDER_WEIGHT, weight_file)))
 
-    # start training
+    # Start training.
     train(metadata, args.out_directory, args.verbose, weight_file, meta_file)
 
 
 def handle_arguments(cmd_line_args=None):
-    """Run training. command-line args may be passed in as a list
+    """Run training. command-line args may be passed in as a list of strings.
     """
 
     import argparse
     parser = argparse.ArgumentParser(description='Perform supervised training on a policy network.')
-    # subparser is always first argument
-    subparsers = parser.add_subparsers(help='sub-command help')
 
-    # sub parser start training
+    # Subparser is always first argument.
+    subparsers = parser.add_subparsers(help='sub-command help', dest='mode')
+
+    # Add 'global' arguments, i.e. those used for both 'train' and 'resume'
+    parser.add_argument("--verbose", "-v", help="Turn on verbose mode", default=False, action="store_true")  # noqa: E501
+
+    # Sub parser for starting training.
     train = subparsers.add_parser('train', help='Start or resume supervised training on a policy network.')  # noqa: E501
-    # required arguments
+    # Required arguments.
     train.add_argument("out_directory", help="directory where metadata and weights will be saved")  # noqa: E501
     train.add_argument("model", help="Path to a JSON model file (i.e. from CNNPolicy.save_model())")  # noqa: E501
     train.add_argument("train_data", help="A .h5 file of training data")
-    # frequently used args
-    train.add_argument("--verbose", "-v", help="Turn on verbose mode", default=False, action="store_true")  # noqa: E501
-    train.add_argument("--minibatch", "-B", help="Size of training data minibatches. Default: " + str(DEFAULT_BATCH_SIZE), type=int, default=None)  # noqa: E501
-    train.add_argument("--epochs", "-E", help="Total number of iterations on the data. Default: " + str(DEFAULT_EPOCH), type=int, default=None)  # noqa: E501
+    # Frequently used args.
+    train.add_argument("--minibatch", "-B", help="Size of training data minibatches. Default: " + str(DEFAULT_BATCH_SIZE), type=int, default=DEFAULT_BATCH_SIZE)  # noqa: E501
```

Comment:
I know the argparse is really useful also with the default values etc but i could not find a way to distinguish user input from default value, to do so i set all default values to `None` and set them to default when validating args.

default batch size is 16 if i am training with batch size 32 and i resume training and setting a new lr i have to input both batch-size and lr because if i leave the batch-size away i will get a confirm promt since the default value will be 16